### PR TITLE
[UI] Converted UserPreferences and preferences form class-based to fun…

### LINKED ISSUE #8687: [UI] Convert  `UserPreferences` and `preferences` form class-based to functional component
### Current Behavior
 `UserPreferences` and `preferences` are class-based  component  

### Desired Behavior
 `UserPreferences` and `preferences` should be a functional component  

### Implementation
Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

### Mockups

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by mukeshkuiry**
…ctional component

**Notes for Reviewers**

This PR fixes #8687

Video Demo : 

https://github.com/meshery/meshery/assets/99157367/b6e506b3-1555-454f-bd35-b4102d409cf4

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by mukeshkuiry** [comment]
@Rajdip019

File: ui/components/UserPreferences.js [hunk 1]
```diff
@@ -154,102 +151,108 @@ function ThemeToggler({ theme, themeSetter, notify, classes }) {
...
+  };
 
-  handleError = (msg) => () => {
-    this.props.updateProgress({ showProgress : false });
-    const notify = this.props.notify;
-    notify({ message : msg, event_type : EVENT_TYPES.ERROR })
-  }
+  const handleError = (name) => () => {
+    props.updateProgress({ showProgress : false });
+    const notify = props.notify;
```
Comment (Rajdip019) [inline ui/components/UserPreferences.js:200]: In all over the codebase we can use `useNotification` hook with a functional component. And remove the `withNotify` wrapper as well.

**Comment by Rajdip019** [comment]
Also @mukeshkuiry you can check on this if the number of `props` is less and the use of those props is significant in the component we can also destructure some of them for easier use later on.

File: ui/components/UserPreferences.js (hunk 1 above)
Comment (mukeshkuiry) [reply ui/components/UserPreferences.js:200]: Sure will update that 👌

**Comment by mukeshkuiry** [comment]
> Also @mukeshkuiry you can check on this if the number of `props` is less and the use of those props is significant in the component we can also destructure some of them for easier use later on.

Okay, there is the use of only two props in preference.js so I will de-structure it.
And in userPreference.js have more than 6 so I guess it's better now

**Comment by mukeshkuiry** [comment]
> > Also @mukeshkuiry you can check on this if the number of `props` is less and the use of those props is significant in the component we can also destructure some of them for easier use later on.
> 
> Okay, there is the use of only two props in preference.js so I will de-structure it. And in userPreference.js have more than 6 so I guess it's better now

@Rajdip019 There are 6 different props used in `preference.js` and 15+ in `usePreference.js` so I guess it's better to don't de-structure it.

**Comment by Rajdip019** [comment]
Cool make the final changes and push a commit.

**Comment by mukeshkuiry** [comment]
@Rajdip019 Done with the necessary changes

**Review by Rajdip019 — APPROVED** [review]

**Review by Rajdip019 — APPROVED** [review]

**Review by Rajdip019 — APPROVED** [review]

**Review by Rajdip019 — APPROVED** [review]
LGTM

**Review by aabidsofi19 — APPROVED** [review]
LGTM

**Pull request merged**